index.js

```diff
@@ -583,6 +626,11 @@
     return typeof other === typeof this && other.valueOf() === this.valueOf();
   }
 
+  //  String$prototype$lte :: String ~> String -> Boolean
+  function String$prototype$lte(other) {
+    return typeof other === typeof this && this.valueOf() <= other.valueOf();
+  }
```

Comment:
@safareli, is this the `typeof` check you were referring to? If so, the one in `String$prototype$equals` would appear to be unnecessary as well.